assignments/closure.js

```diff
@@ -1,19 +1,108 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
+let myChickens = 5;
 
 
-// ==== Challenge 2: Create a counter function ====
-const counter = () => {
-  // Return a function that when invoked increments and returns a counter variable.
+const chickens = () => {
+let totalEggs = 0;
+  return {
+    perDay : () => {
+      totalEggs = totalEggs + 12;
+      return totalEggs;
+    }
+
+  }
+
 };
+
+let egg = chickens();
+// console.log(egg.perDay()
+console.log(`\nEach hen lays ${egg.perDay()} per day. \nI have ${myChickens} chickens, so I collect ${egg.perDay() * myChickens} eggs per day!\n`);
+
+
+
+// ==== Challenge 2: Create a counter function ====
+// Return a function that when invoked increments and returns a counter variable.
 // Example usage: const newCounter = counter();
 // newCounter(); // 1
 // newCounter(); // 2
 
+const cows = () => {
+  let count = 80;
+      return {
```

Comment:
For this one we want to return a function, not an object.

So the function you have for the value of `add` in the returned object, you could just return that! 
```
return () => {
  count++;
  return count;
}
```
Then you would be able to invoke `cow()` on line 41 instead of `cow.add()`. This is because the value of `cow` will be the function returned after invoking `cows()`.